core/testcontainers/core/utils.py

```diff
@@ -53,6 +54,43 @@ def inside_container() -> bool:
     return os.path.exists("/.dockerenv")
 
 
+def create_connection_string(
+    dialect: str,
+    host: str,
+    port: Union[str, int, None] = None,
+    username: Optional[str] = None,
+    password: Optional[str] = None,
+    driver: Optional[str] = None,
+    dbname: Optional[str] = None,
+) -> str:
+    """
+    Returns a connection URL following the RFC-1738 format.
+    Compatible with database clients such as SQLAlchemy and other popular database client libraries.
+
+    Example: postgres+psycopg2://myuser:mypassword@localhost:5432/mytestdb
```

Comment:
It would be nice to make "dialect: str | None " and document that setting it to None or "" will remove the dialect from the URL (psycopg v3 does need a url without dialect!)

Reply:
I would actually split this into a generic "create_connection_url" (`scheme` instead of `dialect` and `driver`, `path` instead of `dbname`) and one call (`create_db_api_connection_url`?)  which takes db api inputs (like the current one) and calls `create_connection_url` with the corresponding arguments.